src/frontend/src/pages/myfeeds.js

```diff
@@ -15,94 +34,312 @@ const useStyles = makeStyles((theme) => ({
+                </em>
+              </Typography>
+            </Box>
+          </Popover>
+        </div>
+      )}
+    </PopupState>
+  );
+}
+
+function DeleteDialogButton({ feed }) {
+  const classes = useStyles();
+  const [open, setOpen] = React.useState(false);
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const removeFeed = () => {
+    console.log(`Removing feed hosted at URL ${feed.url}`);
+    // TODO
+    window.location.reload(false);
+  };
+
+  return (
+    <div>
+      <IconButton classes={{ root: classes.button }} onClick={handleClickOpen}>
+        <Delete color="secondary" />
+      </IconButton>
+      <Dialog
+        open={open}
+        onClose={handleClose}
+        aria-labelledby="alert-dialog-title"
+        aria-describedby="alert-dialog-description"
+      >
+        <DialogTitle id="alert-dialog-title">{`Remove feed hosted at ${feed.url}?`}</DialogTitle>
+        <DialogContent>
+          <DialogContentText id="alert-dialog-description">
+            Telescope will no longer display blog posts from this feed.
+          </DialogContentText>
+        </DialogContent>
+        <DialogActions>
+          <Button onClick={handleClose} color="secondary" variant="outlined" autoFocus>
+            Cancel
+          </Button>
+          <Button
+            onClick={() => {
+              handleClose();
+              removeFeed();
+            }}
+            color="primary"
+            variant="contained"
+          >
+            Confirm
+          </Button>
+        </DialogActions>
+      </Dialog>
+    </div>
+  );
+}
+
+DeleteDialogButton.propTypes = {
+  feed: PropTypes.object,
+};
+
 export default function MyFeeds() {
   const classes = useStyles();
-  const [feedAuthor, setFeedAuthor] = useState('');
-  const [feedUrl, setFeedUrl] = useState('');
+  const [newFeedAuthor, setNewFeedAuthor] = useState('');
+  const [newFeedUrl, setNewFeedUrl] = useState('');
+  const [submitStatus, setSubmitStatus] = useState({ message: '', isError: false });
+  const [userInfo, setUserInfo] = useState({});
+  const [feedHash, updateFeedHash] = useState({}); // { id1: {author: '...', url: '...'} }, ... ]
+
   const { telescopeUrl } = useSiteMetadata();
 
-  function handleAuthorChange(author) {
-    setFeedAuthor(author);
-  }
+  const newFeedAuthorRef = React.createRef();
+  const newFeedUrlRef = React.createRef();
+
+  async function hashUserFeeds() {
+    const [user, feedItems] = await Promise.all([
+      fetch(`${telescopeUrl}/user/info`).then((res) => res.json()),
+      fetch(`${telescopeUrl}/feeds`).then((res) => res.json()),
+    ]);
+    const allFeeds = await Promise.all(
+      feedItems.map((item) => fetch(`${telescopeUrl}${item.url}`))
+    ).then((responses) => Promise.all(responses.map((res) => res.json())));
+
+    if (allFeeds.length && user) {
```

Comment:
This needs to happen at the server.  I know why you're doing it here for testing, but let's not land this.

Reply:
> We need the stuff that @Grommers00 and @c3ho are doing to get "my" feeds vs. all.
> This needs to happen at the server. I know why you're doing it here for testing, but let's not land this.

Agree 100%. I hope @Grommers00 and @c3ho may assist me with developing web API routes (if none exist already) such that the My Feeds component can make use of their work.

Reply:
I've supplemented my makeshift `GET /feeds` solution with a fetch to `GET /user/feeds` in https://github.com/Seneca-CDOT/telescope/pull/933/commits/79712f5943c91d678d2e775750abfd8f9762fe88:

https://github.com/Seneca-CDOT/telescope/blob/79712f5943c91d678d2e775750abfd8f9762fe88/src/frontend/src/pages/myfeeds.js#L57-L90